ProductsManegements/Inventory.cs

```diff
@@ -0,0 +1,198 @@
+------------------------
+");
+                    int i = 1;
+                    foreach (PropertyInfo p in product.GetType().GetProperties())
+                    {
+                        if (p.Name == "Id") continue;
+                        Console.WriteLine($"{i}.Edit {p.Name}");
+                        i++;
+                    }
+                    Console.WriteLine("0.Finish editing ");
+                }
+                while (true)
+                {
+                    menu();
+                    if (int.TryParse(Console.ReadLine(), out input))
+                    {
+                        switch (input)
+                        {
+                            case 1:
+                                string name = Console.ReadLine() ?? string.Empty;
+                                product.Name = name;
+                                Console.ForegroundColor = ConsoleColor.Green;
+                                Console.WriteLine("Product name edited successfully! ");
+                                Console.ResetColor();
+                                break;
+
+                            case 2:
+                                decimal price;
+                                do
+                                {
+                                    Console.WriteLine("Enter valid price: ");
+
+                                } while (!decimal.TryParse(Console.ReadLine(), out price));
+                                product.Price = price;
+                                Console.ForegroundColor= ConsoleColor.Green;
+                                Console.WriteLine("Price edited successfully! ");
+                                Console.ResetColor();
+
+                                break;
+
+                            case 3:
+                                int quantity;
+                                do
+                                {
+                                    Console.WriteLine("Enter the quantity of the product: ");
+
+                                } while (!int.TryParse(Console.ReadLine(), out quantity));
+                                product.Quantity = quantity;
+                                Console.ForegroundColor = ConsoleColor.Green;
+                                Console.WriteLine("Quantity edited successfully! ");
+                                Console.ResetColor();
+                                break;
+                            case 0:
+                                return;
+
+                            default:
+
+                                Console.WriteLine("Not Valid");
+                                break;
+
+
+                        }
+
+                    }
+                }
+            }
+            else
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("No such product! Recheck the name and try again!");
+                Console.ResetColor();
+            }
+
+
+        }
+
+
+    }
+}
```

Comment:
for the Inventory class, I recommend separating the business logic from the logic that handles the display and user interaction. The main idea is to not make the core functionality tightly coupled with how to present the data to the user. For example, you can create two classes:

1. Inventory class which handles all the business logic.
2. InventoryUI class which handles all the user interactions.
```
public class InventoryUI
{
    private Inventory _inventory;

    public InventoryUI(Inventory inventory)
    {
       ...........
}
```